src/oss-characteristics/CharacteristicTool.cs

```diff
@@ -310,6 +326,42 @@ public async Task RunAsync(Options options)
                                 var encodedName = Uri.EscapeUriString(target ?? "unknown");
                                 var purl = new PackageURL("generic", encodedName);
                                 AppendOutput(outputBuilder, purl, analysisResults);
+                                finalResults.Add(analysisResults);
+                            }
+                        }
+                        else if (File.Exists(target))
+                        {
+                            Extractor e = new Extractor();
+                            var tempDirectory = Path.Join(Path.GetTempPath(), Guid.NewGuid().ToString());
+                            var extractorOptions = new ExtractorOptions()
+                            {
+                                ExtractSelfOnFail = true,
```

Comment:
@scovetta FYI on this behavior. This means if we think it’s an archive and it fails to extract we return the archive itself. That means that it is likely a binary file based on the header. Not sure what AI does with those or if that’s the intention here. If you pass RE a regular non archive file you just get it back in a stream regardless of this option.

Reply:
Got it, thanks for the clarification!